### PR TITLE
Add support for TextInput clearTextOnFocus and selectTextOnFocus

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

TextInput `clearTextOnFocus` and `selectTextOnFocus` were never supported on MacOS
https://reactnative.dev/docs/textinput#selecttextonfocus
https://reactnative.dev/docs/textinput#cleartextonfocus-ios

https://github.com/microsoft/react-native-macos/blame/e0e598e19c39bd7cb80b2aefd0ac5f19778721aa/Libraries/Text/TextInput/RCTBaseTextInputViewManager.m#LL62

On iOS, `selectTextOnFocus` has a custom implementation of `selectAll` that dispatched a selection.
https://github.com/facebook/react-native/blob/main/Libraries/Text/TextInput/Multiline/RCTUITextView.m#L181-L190
For NSTextInput, `selectAll` works without needing the custom dispatch.

For NSTextField, the default behavior is to select all when focused. This change added support for removing the selection if `selectTextOnFocus` is false.

NOTE: Text selection disappears once a MouseDown is received. This is native text input behavior.

This also fixes a bug w/ MultiLineTextInput where the selection box was retained even after losing focus.

https://user-images.githubusercontent.com/96719/202796169-d04156e3-718a-4fd8-9583-0112e16bcf12.mp4

## Changelog

[macOS] [Fixed] - TextInput support for `clearTextOnFocus` and `selectTextOnFocus`

## Test Plan

https://user-images.githubusercontent.com/96719/202794592-31c02063-1b18-4047-bf99-6413c01b3325.mp4

### CODE REVIEW

**Comment by shwanton** [comment]
Need to fix build error.

CleanShot 2022-11-22 at 15 16 28

**Comment by shwanton** [comment]
Oops, forgot that this still needs to support iOS as well. Will add back the older path and test.

**Comment by shwanton** [comment]
Fixed on iOS

https://user-images.githubusercontent.com/96719/203449900-363ad060-ed39-48c6-89e9-15c567ba9fce.mp4

**Comment by amgleitman** [comment]
I don't see anything regarding `clearTextOnFocus` except enabling it for macOS in RCTBaseTextInputViewManager.m. If this prop just worked as is, can you explicitly mention that?

**Comment by chiuam** [comment]
Pulled your branch and tested the props, `selectTextOnFocus` worked fine on iOS, but not on macOS :/

**Comment by shwanton** [comment]
> Pulled your branch and tested the props, `selectTextOnFocus` worked fine on iOS, but not on macOS :/

There is some nuance to how this works on desktop. Since the text input can get focus when clicked, the section will also disappear if you click into it. 

Slowed down example:
https://user-images.githubusercontent.com/96719/207995918-f9217241-6cee-4ace-96be-8725e3e8b691.mp4

If you tab into either single or multiline text input, it will select the text. Note, you have to reverse-tab to select the multiline text field since the previous text input will capture additional tabs

https://user-images.githubusercontent.com/96719/207995344-d9b6b06f-85d1-4bc6-a8ba-18f78daadfaa.mp4

Clicking on the view, but not focusing the TextInput will also select the text.

https://user-images.githubusercontent.com/96719/207995997-c1af1b2f-0bbb-4916-9e0d-f4e649830493.mp4

**Comment by Saadnajmi** [comment]
@shwanton by the way, you can use Ctrl+Tab to tab in / out of text fields that capture tab

**Review by chiuam — APPROVED** [review]

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m [hunk 1]
```diff
@@ -436,7 +440,7 @@ - (void)selectAll:(id)sender
 {
   [super selectAll:sender];
 
-#if !TARGET_OS_OSX // TODO(macOS GH#774) For `selectTextOnFocus` prop, which isn't supported on macOS atm.
+#if !TARGET_OS_OSX
```
[the remark this answers is not in the document]
Comment (Saadnajmi) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m:443]: Ditto. Should this come back?

[the remark this answers is not in the document]
Comment (Saadnajmi) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m:443]: Also line 449

**Review by Saadnajmi — COMMENTED** [review]
Apart from tags, LGTM!

File: Libraries/Text/TextInput/Multiline/RCTUITextView.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (Saadnajmi) [reply Libraries/Text/TextInput/Multiline/RCTUITextView.m]: Ended up just pushing the fix myself. Resolving.

**Pull request merged**